src/session.rs

```diff
@@ -435,11 +391,73 @@ impl Session {
                 Ok(Client {
                     tx,
                     is_legacy: true,
+                    _marker: Default::default(),
                 })
             }
             Err(e) => Err(e),
         }
     }
+    fn shutdown(&mut self, ack: Option<Ack>) {
```

Comment:
Could you please move this and `map_handshake_response` back up above to where they are on `master` just so that the diff is easier to read?